nodes/src/nodes/dataset_cobalt/dataset_loader.py

```
@@ -0,0 +1,294 @@
+    before normalisation (they would be resolved away by realpath).
+    """
+    if '..' in path.replace('\\', '/').split('/'):
+        raise ValueError(f'Path traversal detected in file path: {path}')
+
+    real_path = os.path.realpath(path)
+    real_cwd = os.path.realpath(os.getcwd())
+
+    try:
+        common = os.path.commonpath([real_path, real_cwd])
+    except ValueError:
+        raise ValueError(f'Path traversal detected: {path}')
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)